### PR TITLE
Dropdown callbacks WIP

### PR DESCRIPTION
**Pull request by Benqstanley**
Add an onClosed callback for dropdownButtons. This callback is fired whether a new selection is made or not, it is fired after onChanged when a new selection has been made.

*Replace this paragraph with a list of issues related to this PR from our [issue database]. Indicate, which of these issues are resolved or fixed by this PR. There should be at least one issue listed here.*

## Tests

I added the following tests:

"DropdownButton onClosed callback is called when defined" in flutter/test/material/dropdown_test.dart

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Benqstanley** [comment]
I saw two Golden test failures and am unsure how to proceed. Can someone offer guidance? 

@Hixie

**Comment by dkwingsmt** [comment]
Hixie is be available through github pings, but you can find him on discord.
In the meantime, I don't see any golden failures anymore.

File: packages/flutter/lib/src/material/dropdown.dart [hunk 1]
```diff
@@ -895,6 +896,12 @@ class DropdownButton<T> extends StatefulWidget {
   /// The callback will not be invoked if the dropdown button is disabled.
   final VoidCallback onTap;
 
+  ///Called when the dropdown is closed, regardless of if a selection was made.
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/material/dropdown.dart:899]: This and all the docs
```suggestion
  /// Called when the dropdown is closed, regardless of if a selection was made.
```

**Review by dkwingsmt — CHANGES_REQUESTED** [review]
I'm wondering if the better API would be `onCanceled` that is only triggered when the menu is closed _without_ selecting any item. If we have `onCanceled` then it's easy to combine them into an `onClosed`, but it's not easy to separate an `onCanceled` out of `onClosed`.

**Comment by dkwingsmt** [comment]
Hi. Can you update this PR as the review comments? Most of them are trivial, and once they're fixed we can get this PR merged :)

**Comment by dkwingsmt** [comment]
We'll close this PR for now. Feel free to re-open in case you're interested in addressing the comments and finishing the PR!

**Pull request closed without merging**